src/server/mod.rs

```diff
@@ -70,14 +71,32 @@ where
             }
         };
 
+        let upgrade_requested = match (req.header(UPGRADE), req.header(CONNECTION)) {
+            (Some(_), Some(upgrade)) if upgrade.as_str().eq_ignore_ascii_case("upgrade") => true,
+            _ => false,
+        };
+
         let method = req.method();
+
         // Pass the request to the endpoint and encode the response.
-        let res = endpoint(req).await?;
+        let mut res = endpoint(req).await?;
+
+        let upgrade_provided = res.status() == StatusCode::SwitchingProtocols && res.has_upgrade();
```

Comment:
is `res.has_upgrade()` done automatically for default endpoints?

Reply:
yes, it defaults to false — that's already on http_types::Response